stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -396,6 +396,85 @@ function _rdiv!(Dc::Diagonal, Db::Diagonal, Da::Diagonal)
 end
 ldiv!(Dc::Diagonal, Da::Diagonal, Db::Diagonal) = Diagonal(ldiv!(Dc.diag, Da, Db.diag))
 
+# optimizations for (Sym)Tridiagonal and Diagonal
+function (\)(D::Diagonal, S::SymTridiagonal)
+    T = typeof(oneunit(eltype(D)) \ oneunit(eltype(S)))
```

Comment:
Maybe we can use
```julia
# `view(1:10,:)` and `view(1:10,1:10)` has different return. 
(\)(D::Diagonal, S::SymTridiagonal) = D \ Tridiagonal(_evview(S), view(S.dv, 1:length(S.dv)), _evview(S)) 
```
to avoid copy.  (Applies to `/`)